landscaper/common/generate/blueprint/generate.go

```diff
@@ -135,13 +176,78 @@ func cleanDirectory(directory string) error {
 // renderBlueprint creates the blueprint.yaml in the blueprint directory based on the `blueprintTemplate`
 // The blueprint resource must contain fields defining its import fields. Only using a `ref` to a JSONSchema file is not possible.
 // Hence, this function takes the properties of the root schema and renders them as top-level import fields to the blueprint.
-func renderBlueprint(definition common.OpenAPIDefinition, blueprintTemplate *template.Template, blueprintDirectory string) error {
-	var topLevelFields []BlueprintImport
-	requiredFields := sets.NewString(definition.Schema.SchemaProps.Required...)
+func renderBlueprint(importDefinition common.OpenAPIDefinition, exportDefinition *common.OpenAPIDefinition, blueprintTemplate *template.Template, blueprintDirectory string, exportExecutionFileName *string, containerDeployerName *string) error {
+	importsFields, _, err := getTopLevelFields(importDefinition)
+	if err != nil {
+		return err
+	}
+
+	data := map[string]string{
+		"imports": string(importsFields),
+	}
+
+	if exportDefinition != nil {
+		exportsFields, exportedFields, err := getTopLevelFields(*exportDefinition)
+		if err != nil {
+			return err
+		}
+		data["exports"] = string(exportsFields)
+
+		err = writeExportExecutions(exportedFields, blueprintDirectory, exportExecutionFileName, containerDeployerName)
+		if err != nil {
+			return err
+		}
+	}
+
+	var ccdScript bytes.Buffer
+	if err := blueprintTemplate.Execute(&ccdScript, data); err != nil {
+		return err
+	}
+
+	return os.WriteFile(fmt.Sprintf("%s/blueprint.yaml", blueprintDirectory), ccdScript.Bytes(), 0640)
+}
+
+// writeExportExecutions writes a file to the blueprint filesystem containing the export executions corresponding to the exported fields
+// Export executions are necessary to "tell" the landscaper from where (deployer) the valus for the exported fields are actually coming from
+// the generator assumes the field name of the exported field matches the name of the field actually exported vy the deployer.
+// Hence, if your component writes an export.yaml to the export directory with "myCa": "abc", then the exported field name must be "myCa".
+// Please note: the generator currently does NOT support exports of type target (but can be added if needed)
+func writeExportExecutions(exportedFields []BlueprintField, blueprintDirectory string, exportExecutionFileName *string, containerDeployerName *string) error {
+	if exportExecutionFileName == nil {
+		exportExecutionFileName = pointer.String(defaultExportExecutionFileName)
+	}
+
+	if containerDeployerName == nil {
+		containerDeployerName = pointer.String(defaultContainerDeployerName)
+	}
+
+	fields := make(map[string]string, len(exportedFields))
+	for _, field := range exportedFields {
+		fields[field.Name] = fmt.Sprintf("{{- index .values \"deployitems\" \"%s\" \"%s\" | toYaml | nindent 4 }}", *containerDeployerName, field.Name)
```

Comment:
You can use backticks to avoid escaping.
```suggestion
		fields[field.Name] = fmt.Sprintf(`{{- index .values "deployitems" "%s" "%s" | toYaml | nindent 4 }}`, *containerDeployerName, field.Name)
```